fitlins/workflows/base.py

```diff
@@ -150,160 +76,134 @@ def join_dict(base_dict, dict_list):
-    contrast_pattern = '[sub-{subject}/][ses-{session}/]' \
-        '[sub-{subject}_][ses-{session}_]task-{task}[_acq-{acquisition}]' \
-        '[_rec-{reconstruction}][_run-{run}][_echo-{echo}]_bold' \
-        '[_space-{space}]_contrast-{contrast}_{type<effect|stat>}.nii.gz'
-    ds_l1_contrast_maps = pe.Node(
-        BIDSDataSink(base_directory=out_dir,
-                     path_patterns=contrast_pattern),
-        run_without_submitting=True,
-        name='ds_l1_contrast_maps')
-
-    ds_l2_contrast_maps = pe.Node(
-        BIDSDataSink(base_directory=out_dir,
-                     path_patterns=contrast_pattern),
-        run_without_submitting=True,
-        name='ds_l2_contrast_maps')
-
-    # Images
-    image_pattern = '[sub-{subject}/][ses-{session}/]' \
-        '[sub-{subject}_][ses-{session}_]task-{task}[_acq-{acquisition}]' \
-        '[_rec-{reconstruction}][_run-{run}][_echo-{echo}]_bold_' \
-        '{type<design|corr|contrasts>}.svg'
-    ds_design = pe.MapNode(
-        BIDSDataSink(base_directory=out_dir, fixed_entities={'type': 'design'},
-                     path_patterns=image_pattern),
-        iterfield=['entities', 'in_file'],
-        run_without_submitting=True,
-        name='ds_design')
-
-    # ds_corr = pe.MapNode(
-    #     BIDSDataSink(base_directory=out_dir, fixed_entities={'type': 'corr'},
-    #                  path_patterns=image_pattern),
-    #     iterfield=['entities', 'in_file'],
-    #     run_without_submitting=True,
-    #     name='ds_corr')
-
-    # ds_l1_contrasts = pe.MapNode(
-    #     BIDSDataSink(base_directory=out_dir,
-    #                  fixed_entities={'type': 'contrasts'},
-    #                  path_patterns=image_pattern),
-    #     iterfield=['entities', 'in_file'],
-    #     run_without_submitting=True,
-    #     name='ds_l1_contrasts')
-
-    # ds_l2_contrasts = pe.MapNode(
-    #     BIDSDataSink(base_directory=out_dir,
-    #                  fixed_entities={'type': 'contrasts'},
-    #                  path_patterns=image_pattern),
-    #     iterfield=['entities', 'in_file'],
-    #     run_without_submitting=True,
-    #     name='ds_l2_contrasts')
-
-    contrast_plot_pattern = '[sub-{subject}/][ses-{session}/]' \
-        '[sub-{subject}_][ses-{session}_]task-{task}[_acq-{acquisition}]' \
-        '[_rec-{reconstruction}][_run-{run}][_echo-{echo}]_bold' \
-        '[_space-{space}]_contrast-{contrast}_ortho.png'
-    ds_l1_contrast_plots = pe.Node(
-        BIDSDataSink(base_directory=out_dir,
-                     path_patterns=contrast_plot_pattern),
-        run_without_submitting=True,
-        name='ds_l1_contrast_plots')
-
-    ds_l2_contrast_plots = pe.Node(
-        BIDSDataSink(base_directory=out_dir,
-                     path_patterns=contrast_plot_pattern),
-        run_without_submitting=True,
-        name='ds_l2_contrast_plots')
-
-    #
-    # Connections
-    #
+    models = []
+    for ix, step in enumerate(loader.run().outputs.steps):
```

Comment:
Missed this... Either `loader` is in the workflow or it's not. I think maybe it makes more sense to use the `all_models['Steps']` dictionary directly.